daemon/api_apps.go

```diff
@@ -214,8 +214,18 @@ func getLogs(c *Command, r *http.Request, user *auth.UserState) Response {
 		serviceNames[i] = appInfo.ServiceName()
 	}
 
+	// Include journal namespaces if supported. The --namespace option was
+	// introduced in systemd version 245. If systemd is older than that then
+	// we cannot use journal quotas in any case and don't include them.
+	includeNamespaces := false
+	if err := systemd.EnsureAtLeast(245); err == nil {
+		includeNamespaces = true
+	} else if !systemd.IsSystemdTooOld(err) {
+		return InternalError("cannot get systemd version: %v", err)
+	}
+
 	sysd := systemd.New(systemd.SystemMode, progress.Null)
-	reader, err := sysd.LogReader(serviceNames, n, follow)
+	reader, err := sysd.LogReader(serviceNames, n, follow, includeNamespaces)
```

Comment:
nitpick: now that this is not just a trivial 2 lines of calls, maybe this should be a helper in servicestate?

Reply:
Ah yeah that's a nice suggestion, I have moved the functionality to a new function in servicestate, and added unit tests for it.